packages/serverless/src/model/abstract/document.ts

```diff
@@ -12,71 +12,21 @@ import {
   ModelMetadataDatabase,
 } from '@zkdb/storage';
 import { randomUUID } from 'crypto';
-import { ClientSession, Filter, Long, OptionalId } from 'mongodb';
+import { ClientSession, Filter, OptionalId } from 'mongodb';
 import { getCurrentTime } from '../../helper/common.js';
 import logger from '../../helper/logger.js';
 
-/** Database-serialized version of a document record. */
-export type TDocumentRecordSerialized = Omit<
-  TDocumentRecordOptional,
-  'document'
-> & {
-  document: Record<string, TContractSchemaFieldSerializable>;
-};
-
-/** Map of Provable types to their corresponding BSON types. */
-type TProvableSerializationMap = {
-  CircuitString: string;
-  UInt32: Long;
-  Int64: Long;
-  Bool: boolean;
-  PrivateKey: string;
-  PublicKey: string;
-  Signature: string;
-  Character: string;
-  Sign: boolean;
-
-  // Leaving as any as not yet implemented, reason: concerns about indexing,
-  // operators like $gt, $lt and the combination of them (e.g. will $gt on the
-  // indexed field work and how?)
-  Field: any;
-
-  // Leaving as any as not yet implemented, reason: won't fit in bson 64-bit
-  // integer
-  UInt64: any;
-
-  // Leaving as any as not yet implemented
-  MerkleMapWitness: any;
-};
-
-/**
- * Represents a field with a name, kind, and the actual value that can be stored
- * in the database. Rendered as a union of all possible field types.
- */
-export type TContractSchemaFieldSerializable = {
-  [K in TProvableTypeString]: {
-    name: string;
-    kind: K;
-    value: TProvableSerializationMap[K];
-  };
-}[TProvableTypeString];
-
 /**
  * ModelDocument is a class that extends ModelBasic.
  * ModelDocument handle document of zkDatabase with index hook.
  */
 export class ModelDocument extends ModelBasic<
```

Comment:
we adjust to use extends ModelGeneral. it have built in crud method

Reply:
fixed